crates/datasources/src/excel/table.rs

```diff
@@ -33,32 +45,26 @@ pub async fn excel_streaming_table(
         Cow::Owned(first.clone())
     });
 
-    let working_sheet_range = match workbook.worksheet_range(&sheet) {
-        Ok(r) => Ok(r),
-        Err(_) => Err(ExcelError::Load {
-            msg: "Failed to open .xlsx file.".to_owned(),
-        }),
-    };
-
-    let range = working_sheet_range?;
+    let range = workbook
+        .worksheet_range(&sheet)
+        .map_err(|e| Into::<ExcelError>::into(e))?;
```

Comment:
You might be able to get `Ok(workbook.worksheet_range(&sheet)?)` to get the conversion, right? I mgiht be wrong about this case.